crates/dslab-mp/src/mc/strategy.rs

```diff
@@ -0,0 +1,209 @@
+use std::collections::HashMap;
+
+use colored::*;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use crate::mc::events::McEvent::{MessageReceived, TimerFired};
+use crate::mc::events::{DeliveryOptions, McEvent, McEventId};
+use crate::mc::system::McSystem;
+use crate::message::Message;
+use crate::util::t;
+
+#[derive(Clone, PartialEq)]
+pub enum LogMode {
+    Default,
+    Debug,
+}
+
+pub enum LogContext {
+    Default,
+    Dropped,
+    Duplicated,
+    Corrupted,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct McSummary {
+    pub(crate) states: HashMap<String, u32>,
+}
+
+pub trait Strategy {
+    fn run(&mut self, system: &mut McSystem) -> Result<McSummary, String>;
+
+    fn search_step_impl(&mut self, system: &mut McSystem) -> Result<(), String>;
+
+    fn process_event(&mut self, system: &mut McSystem, event_num: usize) -> Result<(), String> {
```

Comment:
```suggestion
    fn process_event(&mut self, system: &mut McSystem, event_id: usize) -> Result<(), String> {
```
и в остальных местах аналогично

Reply:
готово